File: app/src/main/java/com/davidbargados/andrevinadavidbargados/HallOfFame.java
package com.davidbargados.andrevinadavidbargados;

import android.os.Bundle;
import android.widget.TableLayout;
import android.widget.TableRow;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.ArrayList;

public class HallOfFame extends AppCompatActivity {


    private TableLayout tl;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //EdgeToEdge.enable(this);
        setContentView(R.layout.activity_hall_of_fame);
        tl = findViewById(R.id.taulaRankings);

        ArrayList<Integer> ranking = MainActivity.ranking;

        for (Integer cnt : ranking){
            TableRow tr = new TableRow(this);
            TextView tv = new TextView(this);
            tv.setText(String.valueOf(cnt));
            tr.addView(tv);
            tl.addView(tr);

        }

        /*ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });*/

    }
}

File: app/src/main/java/com/davidbargados/andrevinadavidbargados/HallOfFame.kt
package com.davidbargados.andrevinadavidbargados

import android.os.Bundle
import android.widget.Button
import android.widget.TableLayout
import android.widget.TableRow
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class HallOfFame : AppCompatActivity() {
    private lateinit var tl: TableLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hall_of_fame)

        tl = findViewById(R.id.taulaRankings)

        val ranking = MainActivity.ranking

        val headerRow = TableRow(this)
        val headerNom = TextView(this)
        val headerPunts = TextView(this)

        headerNom.text = "Nom"
        headerPunts.text = "Intents"

        headerNom.setPadding(24, 16, 24, 16)
        headerPunts.setPadding(24, 16, 24, 16)

        headerRow.addView(headerNom)
        headerRow.addView(headerPunts)
        tl.addView(headerRow)

        // Rellenamos las filas del ranking
        for (jugador in ranking) {
            val fila = TableRow(this)

            val tvNom = TextView(this)
            val tvPunts = TextView(this)

            tvNom.text = jugador.nom
            tvPunts.text = jugador.puntuacio.toString()

            tvNom.setPadding(24, 16, 24, 16)
            tvPunts.setPadding(24, 16, 24, 16)

            fila.addView(tvNom)
            fila.addView(tvPunts)

            tl.addView(fila)
        }

        val btnTornar = findViewById<Button>(R.id.btnTornar)
        btnTornar.setOnClickListener {
            finish()
        }
    }
}
